Client/core/CVersionUpdater.cpp

```diff
@@ -3377,3 +3378,23 @@ int CVersionUpdater::DoPollPost()
     }
     return RES_OK;
 }
+
+// Issue: #1708
+// Set the update settings to stable build if the user is on 1.5.8.
+// We do this only once, this is why we need to write to registry.
+void CVersionUpdater::OnMainMenuFullyVisible()
+{
+    const SString requiredValue = "discord-rich-presence";
+    const SString lastSubKey = "version-revert-reason";
```

Comment:
Nit: maybe we should start thinking about preferring `std::string`?

Reply:
Indeed, but in this case you aren't using any of those features.

Reply:
Can't we somehow modify `std::string` instead?

Reply:
Maybe we should always use SString instead of std::string for consistency

Reply:
My main problem with `SString` is the lack of modern features (like std::string_view ctor, and as return value), also I dont like the inconsistencies in the naming of the functions, it looks really funny..
```cpp
SString str;
str.erase();
str.Split()
``` 
It just is a pain in the ass to maintain imho, and we should find a way to get rid of it.

Reply:
This is just a nit, so feel free to stay with `SString` @saml1er.

There's some contention here and this isn't the place for discussing / debating `SString` vs `std::string(_view)`, so maybe we can create an issue about it and discuss it there.

(Until an issue is created, if you're curious about why I'm suggesting to prefer `std::string(view)` where possible, [you might find this conversation interesting](https://discord.com/channels/278474088903606273/366384007535001612/740682668647252038). In the meantime, I don't think we should discuss it further in this PR.)

Reply:
> (Until an issue is created, if you're curious about why I'm suggesting to prefer std::string(view) where possible, you might find this conversation interesting. In the meantime, I don't think we should discuss it further in this PR.)

@qaisjp Thanks, that explains it well. Btw, we should stick with one and only use `SString` if std::string doesn't suffice. I'll definitely use std::string for this PR. I think it would be nice if we can add this to the coding guidelines, so other contributors know what to use.